### DISCUSSION TITLE
k_hop_subgraph() not giving all subgraph edges

### DISCUSSION THREAD

**Discussion by yashpaneliya** [pyg-team/pytorch_geometric#8306] [Q&A]
I am new to the graph neural network domain and trying my hands on basics. I was trying to generate a k-hop subgraph using `k_hop_graph()` method. I am not getting the desired output for it.

```python
edge_index = torch.tensor([[0, 0, 0, 0, 1, 2, 3, 4, 5, 6, 7, 8],
                            [1, 2, 3, 4, 5, 6, 7, 8, 12, 9, 10, 11]])

subset, one_hop_edge, mapp, one_mask = k_hop_subgraph(2, 1, edge_index=edge_index)

print(one_hop_edge)
```

The desired output should be the edges `[[0,2], [2,6]]` but I'm getting only `[2,6]`. Can someone help me with this?

**Comment by xnuohz** [comment]
Your output is not right, it should be only `[0, 2]`. Node 6 cannot reach node 2 in your edges, I think `torch_geometric.utils.to_undirected` may be what you want.
